### ISSUE TITLE
Setting route prefix while registering the plugin

### ISSUE THREAD

**Issue by imatefx** [jedireza/generator-hapi-style#20]
Hi
I added the plugin to my project through the manifest.js files plugins object
```
 plugins: {
        './server/api/index': {},
        inert: [{
            select: 'api',
            options: {}
        }],
        vision: [{}],
        lout: {},
        'user-auth': {
            options: {
                routes: {
                    prefix: '/auth'
                }
            }
        }
    }
```
But the plugins prefix is not applied to the plugins route

Is this the correct place to register the plugins and plugin route prefix??

**Comment by jedireza** [comment]
Thanks for opening an issue.

> > I added the plugin to my project through the manifest.js files plugins object

Which plugin? `user-auth`? If so where can I see the code for that?

> Is this the correct place to register the plugins and plugin route prefix??

Plugins yes. I'm not sure about plugin prefix though.

**Comment by jedireza** [comment]
http://hapijs.com/api#serverregisterplugins-options-callback

The object value of the plugin key is the options object. I assume you could do something like:

```js
        'user-auth': {
            routes: {
                prefix: '/auth'
            }
        }
```

But you have to test that out to see if it works.

**Comment by imatefx** [comment]
user-auth plugin is just an example

```
exports.register = function (server, options, next) {

    var api = server.select('api');

    api.route({
        method: 'GET',
        path: '/hello',
        config: {
            handler: showHello,
            description: 'Say hello!',
            notes: 'The user parameter defaults to \'stranger\' if unspecified',
            tags: ['api', 'greeting']
        }
    });
    next();
};

exports.register.attributes = {
    pkg: require('./package.json')
};

var showHello = function (request, reply) {

    reply('Hello World');
};
```

**Comment by imatefx** [comment]
```
   'user-auth': {
            routes: {
                prefix: '/auth'
            }
        }
```
That didn't work

**Comment by jedireza** [comment]
Are you expecting the final route to be `/auth/hello`?

**Comment by imatefx** [comment]
yes

**Comment by imatefx** [comment]
can you create a https://gitter.im account for this repo

**Comment by jedireza** [comment]
I want to do the route prefixing in Frame and Aqua too. I'll keep you posted on my progress.

**Comment by imatefx** [comment]
ok thanks

**Comment by jedireza** [comment]
Ok I found the answer we were looking for. We have to use the array syntax for Glue to pass the right options to the right functions.

```js
'user-auth': [{
    routes: {
        prefix: '/auth'
    },
    options: {
        foo: 'bar'
    },
}]
```

The `options` key here is the `options` argument passed to the plugin registration function.

I've updated my PR in Frame to use this as seen here: https://github.com/jedireza/frame/pull/97/files#diff-36598ac1b037cb0f3cfacf5ef926ee80R50
